fe/fe-core/src/main/java/org/apache/doris/nereids/rules/rewrite/ScalarApplyToJoin.java

```diff
@@ -45,7 +45,7 @@ public class ScalarApplyToJoin extends OneRewriteRuleFactory {
     @Override
     public Rule build() {
         return logicalApply().when(LogicalApply::isScalar).then(apply -> {
-            if (apply.isCorrelated()) {
+            if (apply.isCorrelated() && apply.getCorrelationFilter().isPresent()) {
```

Comment:
maybe when `apply.getCorrelationFilter().isPresent() == false`, apply.isCorrelated() should return `false`?